File: frontend/src/types/post.ts
import { UserShortInfo } from "./user";
import { UserReaction } from "./userReaction";
import { Comment } from "./comment";

export interface Post {
    id: string;
    userId: string;
    title: string;
    content: string;
    createdAt: Date;
    reactionsCount: number;
    user: UserShortInfo;
}
export interface PostInfo {
    id: string;
    userId: string;
    title: string;
    content: string;
    createdAt: Date;
    reactionsCount: number;
    user: UserShortInfo;
    userReactions: UserReaction[];
    comments: Comment[];
}

File: backend/Core/Domain/Entities/Post.cs
using Core.Domain.IdentityEntities;

namespace Core.Domain.Entities
{
    public class Post
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public string? Title { get; set; }
        public string? Content { get; set; }
        public DateTime CreatedAt { get; set; }
        public ApplicationUser? User { get; set; }
        public ICollection<UserReaction>? UserReactions { get; set; }
        public ICollection<Comment>? Comments { get; set; }
    }
}